### PR TITLE
Writes up user scenarios to explain how mvn package would work with Pyrsia

### PR DESCRIPTION
**Pull request by betarelease**
## Description

This PR documents user scenarios for mvn update

## PR Checklist

- [x] I've read the [contributing guidelines](https://github.com/pyrsia/.github/blob/main/contributing.md).
- [x] I've read ["What is a Good PR?"](https://github.com/pyrsia/pyrsia/blob/main/docs/get_involved/good_pr.md)
- [x] I've included a good title and brief description along with how to review them.
- [x] I've linked any associated an [issue](https://github.com/pyrsia/pyrsia/issues).

### CODE REVIEW

**Comment by betarelease** [comment]
@erwin1 I wrote up scenarios here but I had better luck drawing them. So please look at the lucidchart attached as well as this doc.

File: docs/developers/maven_and_pyrsia.md [hunk 1]
```diff
@@ -0,0 +1,15 @@
+# mvn User scenarios
+
+__This assumes that Pyrsia network has a set of well known java binaries(jars) that have been built from source and are available on the cloud node. Adding new java binaries to the Pyrsia network is not in scope for this use case.__
+
+Please refer to [mvn update scenarios](<https://lucid.app/lucidchart/d5254e8f-69c5-49d9-acae-91aff72382e2/edit?viewport_loc=42%2C69%2C1912%2C1107%2C0_0&invitationId=inv_a3c2b95f-f098-4758-8a6d-016f569b572c>#)
+
+- As a user I can use a Java build tool like Maven to download Maven artifacts from the Pyrsia network.
+
+- As a user of maven, I should be able to run `mvn update` and receive the dependencies for my java project from the Pyrsia network instead of the mavencentral repository.
```
Comment (fishseabowl) [inline docs/developers/maven_and_pyrsia.md:9]: "mavencentral repository" or "maven central repository"?

Comment (erwin1) [inline docs/developers/maven_and_pyrsia.md:9]:
```suggestion
- As a user of maven, I should be able to run `mvn update` and receive the dependencies for my java project from the Pyrsia network instead of the maven central repository.
```

Comment (erwin1) [reply docs/developers/maven_and_pyrsia.md]: Sorry, with all these docs, I thought for a minute this was my PR so I went ahead and changed it :)

File: docs/developers/maven_and_pyrsia.md [hunk 2]
```diff
@@ -0,0 +1,15 @@
+# mvn User scenarios
+
+__This assumes that Pyrsia network has a set of well known java binaries(jars) that have been built from source and are available on the cloud node. Adding new java binaries to the Pyrsia network is not in scope for this use case.__
+
+Please refer to [mvn update scenarios](<https://lucid.app/lucidchart/d5254e8f-69c5-49d9-acae-91aff72382e2/edit?viewport_loc=42%2C69%2C1912%2C1107%2C0_0&invitationId=inv_a3c2b95f-f098-4758-8a6d-016f569b572c>#)
+
+- As a user I can use a Java build tool like Maven to download Maven artifacts from the Pyrsia network.
+
+- As a user of maven, I should be able to run `mvn update` and receive the dependencies for my java project from the Pyrsia network instead of the maven central repository.
```
[the remark this answers is not in the document]
Comment (betarelease) [reply docs/developers/maven_and_pyrsia.md:9]: indeed I learnt that about maven recently. I will update.

[the remark this answers is not in the document]
Comment (betarelease) [reply docs/developers/maven_and_pyrsia.md:9]: done.

[the remark this answers is not in the document]
Comment (betarelease) [reply docs/developers/maven_and_pyrsia.md:9]: updated.

File: docs/developers/maven_and_pyrsia.md [hunk 3]
```diff
@@ -1,14 +1,19 @@
...
+__This assumes that Pyrsia network has a set of well known java binaries (jars) that have been built from source and are available on the cloud node. Adding new java binaries to the Pyrsia network is not in scope for this use case.__
 
-Please refer to [mvn update scenarios](<https://lucid.app/lucidchart/d5254e8f-69c5-49d9-acae-91aff72382e2/edit?viewport_loc=42%2C69%2C1912%2C1107%2C0_0&invitationId=inv_a3c2b95f-f098-4758-8a6d-016f569b572c>#)
+Please refer to [mvn scenarios](<https://lucid.app/lucidchart/d5254e8f-69c5-49d9-acae-91aff72382e2/edit?viewport_loc=42%2C69%2C1912%2C1107%2C0_0&invitationId=inv_a3c2b95f-f098-4758-8a6d-016f569b572c>#)
 
 - As a user I can use a Java build tool like Maven to download Maven artifacts from the Pyrsia network.
 
-- As a user of maven, I should be able to run `mvn update` and receive the dependencies for my java project from the Pyrsia network instead of the maven central repository.
+- As a user of maven, I should be able to run `mvn commands` and receive the dependencies for my java project from the Pyrsia network instead of the Maven Central Repository.
```
Comment (jperedadnr) [inline docs/developers/maven_and_pyrsia.md:9]: What do you mean by `mvn commands`? There is no such plugin `commands`. Or maybe do you mean "`mvn` commands"?

File: docs/developers/maven_and_pyrsia.md [hunk 4]
```diff
@@ -1,14 +1,19 @@
...
+  - ( We may need to use `mvn -U dependency:resolve` to test this feature. This command will look for the dependencies on the network and should go to Pyrsia before looking for them in Maven Central)
```
Comment (jperedadnr) [inline docs/developers/maven_and_pyrsia.md:10]: Maybe rephrase to:
"  - (For instance, we could use "...

File: docs/developers/maven_and_pyrsia.md [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+
+- As a user I can use a Java build tool like Maven to download Maven artifacts from the Pyrsia network.
+
+- As a user of maven, I should be able to run `mvn commands` and receive the dependencies for my java project from the Pyrsia network instead of the Maven Central Repository.
+  - ( We may need to use `mvn -U dependency:resolve` to test this feature. This command will look for the dependencies on the network and should go to Pyrsia before looking for them in Maven Central)
+  The user needs to configure their repository-order as follows:
+  - A Maven project can contain a number of remote repositories (via pom.xml, super pom, or local/global settings.xml). By default the local repository at ~/.m2 and Maven Central are always included. The precedence of repositories is defined here: <https://maven.apache.org/guides/mini/guide-multiple-repositories.html#repository-order>.
+
+Adding the Pyrsia repository to the pom/settings will allow downloading from Pyrsia first (if the artifact is not already downloaded into .m2, that is), and if not present, then from Maven Central.)
```
Comment (jperedadnr) [inline docs/developers/maven_and_pyrsia.md:14]: into ~/.m2

File: docs/developers/maven_and_pyrsia.md (hunk 3 above)
Comment (betarelease) [reply docs/developers/maven_and_pyrsia.md:9]: Yes I mean maven commands. I will edit that.

File: docs/developers/maven_and_pyrsia.md (hunk 4 above)
Comment (betarelease) [reply docs/developers/maven_and_pyrsia.md:10]: will do

File: docs/developers/maven_and_pyrsia.md (hunk 5 above)
Comment (betarelease) [reply docs/developers/maven_and_pyrsia.md:14]: will update

**Review by jperedadnr — APPROVED** [review]
Looks good!

**Review by erwin1 — APPROVED** [review]
